### ISSUE TITLE
Screen Reader: Sign in: The 'Licenses' link is announced without a role

### ISSUE THREAD

**Issue by olenamihno** [Expensify/App#77544]
## Action Performed:
Prerequisites:
- the user is not signed in

1. Using iOS VoiceOver, open the Expensify app
2. Swipe to the 'Licenses' link
3. Note how the control is announced

## Expected Result:
The link should be announced with a link role.

## Actual Result:
The link is announced without a role.

## Area issue was found in:
Sign in

## Failed WCAG checkpoints
WCAG 4.1.2 Name, Role, Value
https://www.w3.org/WAI/WCAG22/Understanding/name-role-value

## User impact:
Providing role, state, and value information on all user interface components enables compatibility with assistive technology, such as screen readers, screen magnifiers, and speech recognition software, used by people with disabilities.

## Suggested resolution:
Add the link role to announce the control as a link.
https://reactnative.dev/docs/accessibility#accessibilityrole

## Environment:
Apple iPhone 11 - iOS 17.5.1, H0T Mobile - Native (No mobile browser)

## Screenshots/Videos

https://github.com/user-attachments/assets/0309b316-6c1c-47bd-b789-dbe121169f45

<img width="828" height="1792" alt="Image" src="https://github.com/user-attachments/assets/f826208d-3ef1-416a-8e96-e58546ed1626" />

**Version Number:** Production
**Reproducible in staging?:** NA
**Reproducible in production?:** Yes
**Email or phone of affected tester (no customers):** No
**Logs:** https://stackoverflow.com/c/expensify/questions/4856  NA
**Notes/Photos/Videos:** Any additional supporting documentation  Yes
**Expensify/Expensify Issue URL:** https://platform.applause.com/company/12102/products/36628/community-issues/7005375
**Issue reported by:** NA
**Slack conversation:** NA
